python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def _drop_table(self, identifier: Union[str, Identifier], delete_data: bool = False) -> None:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            with self._client as open_client:
+                open_client.drop_table(dbname=database_name, name=table_name, deleteData=delete_data)
+        except NoSuchObjectException as e:
+            # When the namespace doesn't exists, it throws the same error
+            raise NoSuchTableError(f"Table {table_name} cannot be found") from e
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier)
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier, True)
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        from_database_name, from_table_name = self.identifier_to_tuple(from_identifier)
+        to_database_name, to_table_name = self.identifier_to_tuple(to_identifier)
+        try:
+            with self._client as open_client:
+                tbl = open_client.get_table(dbname=from_database_name, tbl_name=from_table_name)
+                tbl.dbName = to_database_name
+                tbl.tableName = to_table_name
+                open_client.alter_table(dbname=from_database_name, tbl_name=from_table_name, new_tbl=tbl)
+        except NoSuchObjectException as e:
+            raise NoSuchTableError(f"Table {from_database_name}.{from_table_name} not found") from e
+        except InvalidOperationException as e:
+            raise NoSuchNamespaceError(f"Destination database {to_database_name} does not exists") from e
+        return Table()
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+
+        hive_database = HiveDatabase(name=database_name, parameters=properties)
+
+        try:
+            with self._client as open_client:
+                open_client.create_database(hive_database)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Database {database_name} already exists") from e
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            with self._client as open_client:
+                open_client.drop_database(database_name, deleteData=True, cascade=False)
+        except InvalidOperationException as e:
+            raise NamespaceNotEmptyError(f"Database {database_name} is not empty") from e
+        except MetaException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} does not exists") from e
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
```

Comment:
In the Java client, the behavior of this method is modified by the `list-all-tables` property. When true, `get_all_tables` is called like in this implementation. By default, Java will check that the `table_type` table parameter is `iceberg` (ignoring case). That's pretty slow so I would probably just move on with this implementation.

I just wanted to note that there could be an expectation for an option to only return Iceberg tables.

Reply:
Wasn't aware of that, thanks. We could make this configurable in the future, and it would be easy to parallelize this.